server/recceiver/cfstore.py

```diff
@@ -355,460 +608,687 @@ def clean_service(self):
+
+    Modifies:
+        channels
+
+    Args:
+        channels: The list of channels.
+        channel_name: The name of the channel.
+        ioc_info: The IOC information.
+        new_properties: The new properties.
+        cf_config: configuration of processor
+        record_info_by_name: The dictionary of record names to information.
+    """
+
+    channels.append(CFChannel(channel_name, ioc_info.owner, new_properties))
+    _log.debug("Add new channel: %s", channel_name)
+    if cf_config.alias_enabled:
+        if channel_name in record_info_by_name:
+            alias_properties = [CFProperty.alias(ioc_info.owner, channel_name)]
+            for p in new_properties:
+                alias_properties.append(p)
+            for alias in record_info_by_name[channel_name].aliases:
+                channels.append(CFChannel(alias, ioc_info.owner, alias_properties))
+                _log.debug("Add new alias: %s from %s", alias, channel_name)
+
+
+def _update_channelfinder(
+    processor: CFProcessor, record_info_by_name: Dict[str, RecordInfo], records_to_delete, ioc_info: IocInfo
+) -> None:
+    """Update Channelfinder with the provided IOC and Record information.
+
+    Calculates the changes required to the channels list and pushes the update the channelfinder.
+
+    Args:
+        processor: The processor.
+        record_info_by_name: The dictionary of record names to information.
+        records_to_delete: The list of records to delete.
+        ioc_info: The IOC information.
+    """
+    _log.info("CF Update IOC: %s", ioc_info)
+    _log.debug("CF Update IOC: %s record_info_by_name %s", ioc_info, record_info_by_name)
     # Consider making this function a class methed then 'processor' simply becomes 'self'
     client = processor.client
-    channels_dict = processor.channel_dict
+    channel_ioc_ids = processor.channel_ioc_ids
     iocs = processor.iocs
-    conf = processor.conf
-    recceiverid = conf.get(RECCEIVERID_KEY, RECCEIVERID_DEFAULT)
-    new_channels = set(recordInfoByName.keys())
-
-    if iocid in iocs:
-        hostName = iocs[iocid]["hostname"]
-        iocName = iocs[iocid]["iocname"]
-        owner = iocs[iocid]["owner"]
-        iocTime = iocs[iocid]["time"]
-        iocIP = iocs[iocid]["iocIP"]
-    else:
-        _log.warning("IOC Env Info not found: {iocid}".format(iocid=iocid))
+    cf_config = processor.cf_config
+    recceiverid = processor.cf_config.recceiver_id
+    new_channels = set(record_info_by_name.keys())
+    iocid = ioc_info.ioc_id
 
-    if hostName is None or iocName is None:
-        raise Exception("missing hostName or iocName")
+    if iocid not in iocs:
+        _log.warning("IOC Env Info %s not found in ioc list: %s", ioc_info, iocs)
 
-    if processor.cancelled:
-        raise defer.CancelledError()
+    if ioc_info.hostname is None or ioc_info.ioc_name is None:
+        raise Exception(f"Missing hostName {ioc_info.hostname} or iocName {ioc_info.ioc_name}")
```

Comment:
Prefer explicit exception. Can maybe use standard lib `AttributeError` or `ValueError` here, if you don't want to create custom one(s).